### PR TITLE
[Core] Replace trivial loops in utilities/auxiliar_model_part_utilities

### PR DESCRIPTION
**Pull request by matekelemen**
**Description**
Replace trivial ```omp``` loops with ```block_for_each``` and handle some others that were tagged for parallelization.

### CODE REVIEW

File: kratos/utilities/auxiliar_model_part_utilities.cpp [hunk 1]
```diff
@@ -189,20 +191,21 @@ void AuxiliarModelPartUtilities::RemoveElementsAndBelongings(Flags IdentifierFla
...
-                }
+        block_for_each(
+            r_array_nodes,
+            [&IdentifierFlag](Node<3>& r_node)
+            { r_node.Set(IdentifierFlag, true); }
+        );
+
+        block_for_each(
+            i_mesh->Elements(),
+            [&IdentifierFlag](ModelPart::ElementType& r_element)
```
Comment (philbucher) [inline kratos/utilities/auxiliar_model_part_utilities.cpp]:
```suggestion
            [&IdentifierFlag](Element& rElement)
```

**Review by loumalouomega — DISMISSED** [review]
Okay

**Comment by philbucher** [comment]
ping @matekelemen

File: kratos/utilities/auxiliar_model_part_utilities.cpp [hunk 2]
```diff
@@ -104,20 +105,21 @@ void AuxiliarModelPartUtilities::RemoveElementAndBelongings(
...
-        it_node->Set(IdentifierFlag, true);
-    }
 
-    // TODO: Add OMP
-    for (auto& elem : mrModelPart.Elements(ThisIndex)) {
-        if (elem.Id() != ElementId) {
-            for (auto& node : elem.GetGeometry()) {
-                node.Set(IdentifierFlag, false);
-            }
+    block_for_each(
```
Comment (philbucher) [inline kratos/utilities/auxiliar_model_part_utilities.cpp]: here you can actually use this method: https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/utilities/variable_utils.h#L815

File: kratos/utilities/auxiliar_model_part_utilities.cpp [hunk 3]
```diff
@@ -189,20 +191,21 @@ void AuxiliarModelPartUtilities::RemoveElementsAndBelongings(Flags IdentifierFla
...
-            it_node->Set(IdentifierFlag, true);
-        }
-
-        // TODO: Add OMP
-        for (auto& elem : i_mesh->Elements()) {
-            if (elem.IsNot(IdentifierFlag)) {
-                for (auto& node : elem.GetGeometry()) {
-                    node.Set(IdentifierFlag, false);
-                }
+        block_for_each(
```
Comment (philbucher) [inline kratos/utilities/auxiliar_model_part_utilities.cpp]: same

File: kratos/utilities/auxiliar_model_part_utilities.cpp [hunk 4]
```diff
@@ -238,20 +241,21 @@ void AuxiliarModelPartUtilities::RemoveElementsAndBelongingsFromAllLevels(Flags
...
-        it_node->Set(IdentifierFlag, true);
-    }
-
-    // TODO: Add OMP
-    for (auto& cond : mrModelPart.Conditions(ThisIndex)) {
-        if (cond.Id() != ConditionId) {
-            for (auto& node : cond.GetGeometry()) {
-                node.Set(IdentifierFlag, false);
-            }
+    block_for_each(
```
Comment (philbucher) [inline kratos/utilities/auxiliar_model_part_utilities.cpp]: same

File: kratos/utilities/auxiliar_model_part_utilities.cpp [hunk 5]
```diff
@@ -323,20 +327,21 @@ void AuxiliarModelPartUtilities::RemoveConditionsAndBelongings(Flags IdentifierF
...
-            it_node->Set(IdentifierFlag, true);
-        }
-
-        // TODO: Add OMP
-        for (auto& cond : i_mesh->Conditions()) {
-            if (cond.IsNot(IdentifierFlag)) {
-                for (auto& node : cond.GetGeometry()) {
-                    node.Set(IdentifierFlag, false);
-                }
+        block_for_each(
```
Comment (philbucher) [inline kratos/utilities/auxiliar_model_part_utilities.cpp]: same

**Review by philbucher — COMMENTED** [review]
some more comments for simplifications

File: kratos/utilities/auxiliar_model_part_utilities.cpp (hunk 2 above)
Comment (matekelemen) [reply kratos/utilities/auxiliar_model_part_utilities.cpp]: As far as I understand, I have to construct an instance of ```VariableUtils``` to call that. Is there any reason for that function to be non-static? It doesn't seem to use any members.

Comment (philbucher) [reply kratos/utilities/auxiliar_model_part_utilities.cpp]: yes exactly
there are no reasons except legacy :)

**Review by philbucher — APPROVED** [review]
thx!

**Pull request merged**